clients/python/girder_client/__init__.py

```diff
@@ -488,6 +488,71 @@ def uploadFileToItem(self, itemId, filepath):
                                 ' not receive object with _id. Got instead: ' +
                                 json.dumps(obj))
 
+    def uploadFile(self, parentId, stream, name, size, parentType=None,
+                   progressCallback=None):
+        """
+        Uploads a file into an item or folder.
+
+        :param parentId: The ID of the folder or item to upload into.
+        :type parentId: str
+        :param stream: Readable stream object.
+        :type stream: file-like
+        :param name: The name of the file to create.
+        :type name: str
+        :param size: The length of the file.
+        :type size: str
+        :param parentType: 'item' or 'folder'. Default assumes parent type is
+            item.
+        :type parentType: str or None
+        :param progressCallback: If passed, will be called after each chunk
+            with progress information. It passes a single parg to the callable
+            which is a dict of information about progress.
+        :type progressCallback: callable
+        :returns: The file that was created on the server.
+        """
+        obj = self.post('file', {
+            'parentType': parentType or 'item',
+            'parentId': parentId,
+            'name': name,
+            'size': size
+        })
+        if '_id' in obj:
+            uploadId = obj['_id']
+        else:
+            raise Exception(
+                'After creating an upload token for a new file, expected '
+                'an object with an id. Got instead: ' + json.dumps(obj))
+
+        offset = 0
+        while True:
+            data = stream.read(self.MAX_CHUNK_SIZE)
+
+            if not data:
+                break
+
+            params = {
```

Comment:
It would eventually be nice to refactor this and logic in ``uploadFileToItem`` to a common place.